packages/@aws-cdk/aws-codedeploy/README.md

```diff
@@ -373,6 +391,218 @@ const application = codedeploy.EcsApplication.fromEcsApplicationName(
 );
 ```
 
+## ECS Deployment Groups
+
+CodeDeploy can be used to deploy to load-balanced ECS services.
+CodeDeploy performs ECS blue-green deployments by managing ECS task sets and load balancer
+target groups.  During a blue-green deployment, one task set and target group runs the
+original version of your ECS task definition ('blue') and another task set and target group
+runs the new version of your ECS task definition ('green').
+
+CodeDeploy orchestrates traffic shifting during ECS blue-green deployments by using
+a load balancer listener to balance incoming traffic between the 'blue' and 'green' task sets/target groups
+running two different versions of your ECS task definition.
+Before deployment, the load balancer listener sends 100% of requests to the 'blue' target group.
+When you publish a new version of the task definition and start a CodeDeploy deployment,
+CodeDeploy can send a small percentage of traffic to the new 'green' task set behind the 'green' target group,
+monitor, and validate before shifting 100% of traffic to the new version.
+
+To create a new CodeDeploy Deployment Group that deploys to a ECS service:
+
+```ts
+declare const myApplication: codedeploy.EcsApplication;
+declare const service: ecs.FargateService;
+declare const blueTargetGroup = elbv2.ITargetGroup;
+declare const greenTargetGroup = elbv2.ITargetGroup;
+declare const listener = elbv2.IApplicationListener;
+
+new codedeploy.EcsDeploymentGroup(stack, 'BlueGreenDG', {
+  services: [service],
+  blueGreenDeploymentConfig: {
+    blueTargetGroup,
+    greenTargetGroup,
+    listener,
+  },
+  deploymentConfig: codedeploy.EcsDeploymentConfig.CANARY_10PERCENT_5MINUTES,
+});
+```
+
+In order to deploy a new task definition version of the ECS service,
+deploy the changes directly through CodeDeploy using the CodeDeploy APIs or console.
+When the `CODE_DEPLOY` deployment controller is used, the ECS service cannot be
+deployed with a new task definition version through CloudFormation.
```

Comment:
Is this feature documented yet? I can't find anything explaining this, the only thing that talks
about doing blue/green deployments through CloudFormation talk about using the CFN hook.

I would also want to know if I'm using the CDK to manage my ECS deployments, how do I deploy code
updates? It doesn't sound like that is possible?

Reply:
Correct, I can add a note about using CFN hooks if the user is looking to do blue-green ECS deployments in CDK, vs just standing up all the resources needed and deploying changes outside the stack